dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -102,7 +102,7 @@ internal object CoreFeature {
         initializeClockSync(appContext)
         setupInfoProviders(appContext, consent)
         setupOkHttpClient(configuration.needsClearTextHttp)
-        firstPartyHostDetector = FirstPartyHostDetector(configuration.firstPartyHosts)
+        firstPartyHostDetector.addKnownHosts(configuration.firstPartyHosts)
```

Comment:
so I guess this is the main reason of the issue?

Reply:
Mostly this one but sometimes our clients can have multiple OkHttp clients and they want to be able to configure separated traced hosts for them. Having this in mind we need a global traced hosts entry + the possibility to add custom traced hosts/RumInterceptor and for that we were using the `localFirstPartyHostDetector` which was not used anymore.